src/cascadia/TerminalSettingsModel/ActionArgsMagic.h

```diff
@@ -166,4 +222,47 @@ public:
         til::hasher h;                                                                            \
         argsMacro(HASH_ARGS);                                                                     \
         return h.finalize();                                                                      \
+    }                                                                                             \
+    winrt::Windows::Foundation::Collections::IVectorView<ArgDescriptor> GetArgDescriptors()       \
+    {                                                                                             \
+        static const auto descriptors = INIT_ARG_DESCRIPTORS(argsMacro);                          \
+        return descriptors;                                                                       \
+    }                                                                                             \
+    IInspectable GetArgAt(uint32_t index) const                                                   \
+    {                                                                                             \
+        X_MACRO_INDEX_BASE();                                                                     \
+        argsMacro(GET_ARG_BY_INDEX) return nullptr;                                               \
+    }                                                                                             \
+    void SetArgAt(uint32_t index, IInspectable value)                                             \
+    {                                                                                             \
+        X_MACRO_INDEX_BASE();                                                                     \
+        argsMacro(SET_ARG_BY_INDEX)                                                               \
+    }
+
+#define PARTIAL_ACTION_ARG_BODY(className, argsMacro)                                       \

```

Comment:
nit please document what is left out of this one

if possible can you make the other macro _use_ this one and then expand on it?

like

```
#define BIGGER_ACTION_ARG_BODY(x, y) \
PARTIAL_ACTION_ARG_BODY(x, y) \
ExtraThing1; \
ExtraThing2;
```

Reply:
(that will help document what's missing from PARTIAL)